.github/workflows/release.yml

```diff
@@ -42,6 +42,38 @@ jobs:
           path: ./dist/*.tar.gz
           retention-days: 5
 
+  test-sdist:
+    name: Test sdist
+    needs: [build-sdist]
+    runs-on: ubuntu-latest
+    container:
+      image: "osgeo/gdal:ubuntu-small-3.4.0" # python 3.8.10
+
+    steps:
+      - name: Download sdist from artifacts
+        uses: actions/download-artifact@v2
+        with:
+          path: wheelhouse
+
+      - name: Install python
+        run: |
+          apt-get update && apt-get install -y python3 python3-pip
+
+      - name: Build from sdist
+        shell: bash
+        run: |
+          python3 -m pip install --no-cache-dir cython~=0.29
+          python3 -m pip install --no-cache-dir --no-build-isolation wheelhouse/artifact/*.tar.gz
```

Comment:
This should also work without the `--no-build-isolation` (and installing cython upfront)? 
(which then also tests that the pyproject.toml build deps are correct)